File: 0908-middle-of-the-linked-list/0908-middle-of-the-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
 #include <cmath>
class Solution {
public:
    ListNode* middleNode(ListNode* head) {
       if(head==NULL) return head;
       ListNode * t = head;
       int l=0;
       while(t!=NULL){
        t=t->next;
        l++;
       }   
       int j = (l/2);
       t=head;
       for(int i=0;i<j;i++)
       {
         t=t->next;
       }
       return t;
    }
};

File: 0908-middle-of-the-linked-list/0908-middle-of-the-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode middleNode(ListNode head) {
        // int c =0;
        // ListNode t = head;
        // while(t!=null)
        // {
        //     c++;
        //     t=t.next;
        // }
        // int m = c/2;
        // t=head;
        // for(int i=0;i<m;i++){
        //     t=t.next;
        // }
        // return t;
        ListNode s = head;
        ListNode t = head;
        while(t!=null && t.next!=null){
            s=s.next;
            t=t.next.next;
        }
        return s;
    }
}